### DISCUSSION TITLE
I plan to make AGE support in node-pg-type.

### DISCUSSION THREAD

**Discussion by shinhanbyeol** [apache/age#1715] [Ideas] [answered]
node js developers is mainly use the client of the project called node-postgres.

The node-postgres is not officially supporting "agtype" (type id 17260).

I think it's a problem that can be solved by supporting "agtype" in a project called pg-type

i going to make a pull request for the pg-type project.
It would be great if members of the AGE community could join us or give us their opinions.

It's a fork branch here.
https://github.com/shinhanbyeol/node-pg-types/tree/age/pg-type-extension-age

The test case code below and
Please refer to the picture that was tested and obtained from the console window
I would appreciate it if you could review the return type for the parser and give me your opinion.

<img width="1103" alt="image" src="https://github.com/apache/age/assets/76985229/e92554c7-8d19-42bd-b900-283936dbf4b4">

<img width="538" alt="image" src="https://github.com/apache/age/assets/76985229/28d08063-016a-4768-8446-8f6bccadebfb">

**Comment by shinhanbyeol** [comment]
@apache/age-committers 
I would appreciate it if someone could suggest a test case to test agetype or contribute to my fork branch.

I don't have enough knowledge about agtype yet, so I'm going to receive more test cases 
I feel like I have to go through it. I hope others can help me.

First of all, I would like to study with the information of this link.
https://age.apache.org/age-manual/master/intro/types.html

**Comment by shinhanbyeol** [comment]
Oh... I'm sorry. @emotionbug didn't know that they had already developed parser within the age project driver/nodejs

Maybe I should've looked it up earlier..

**Reply by emotionbug** [reply, marked as answer]
As everyone knows, the data structure in Apache AGE is not strictly JSON.
it is similar to JSON, but is designed to hold more specialized types.

so, developers wrote this rule as ANTRL Code.
i took it and made it available to JAVA(JDBC) and JavaScript(node-postgres).

i think these things should evolve and become more convenient over time.
so, i think this is a good attempt.

and, also a lot more todo, like fixing the ageviewer... 
( refactor to next.js and typescript.... / version sepcific issues.... )
